subgraphs/rocket-pool/src/updaters/financialMetrics.ts

```diff
@@ -0,0 +1,490 @@
+      : pool.cumulativeTotalRevenueUSD.minus(
+          pool.cumulativeProtocolSideRevenueUSD
+        );
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeSupplySideRevenueUSD =
+    pool.cumulativeSupplySideRevenueUSD;
+  poolMetricsDailySnapshot.dailySupplySideRevenueUSD =
+    poolMetricsDailySnapshot.dailyTotalRevenueUSD <=
+    poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD
+      ? BIGDECIMAL_ZERO
+      : poolMetricsDailySnapshot.dailyTotalRevenueUSD.minus(
+          poolMetricsDailySnapshot.dailyProtocolSideRevenueUSD
+        );
+  poolMetricsDailySnapshot.save();
+
+  // Pool Hourly
+  poolMetricsHourlySnapshot.cumulativeSupplySideRevenueUSD =
+    pool.cumulativeSupplySideRevenueUSD;
+  poolMetricsHourlySnapshot.hourlySupplySideRevenueUSD =
+    poolMetricsHourlySnapshot.hourlyTotalRevenueUSD <=
+    poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD
+      ? BIGDECIMAL_ZERO
+      : poolMetricsHourlySnapshot.hourlyTotalRevenueUSD.minus(
+          poolMetricsHourlySnapshot.hourlyProtocolSideRevenueUSD
+        );
+  poolMetricsHourlySnapshot.save();
+
+  // Protocol
+  protocol.cumulativeSupplySideRevenueUSD = pool.cumulativeSupplySideRevenueUSD;
+  protocol.save();
+
+  // Financial Daily
+  financialMetrics.cumulativeSupplySideRevenueUSD =
+    pool.cumulativeSupplySideRevenueUSD;
+  financialMetrics.dailySupplySideRevenueUSD =
+    financialMetrics.dailyTotalRevenueUSD <=
+    financialMetrics.dailyProtocolSideRevenueUSD
+      ? BIGDECIMAL_ZERO
+      : financialMetrics.dailyTotalRevenueUSD.minus(
+          financialMetrics.dailyProtocolSideRevenueUSD
+        );
+  financialMetrics.save();
+}
+
+export function getOrCreateFinancialDailyMetrics(
+  block: ethereum.Block
+): FinancialsDailySnapshot {
+  const dayId: string = (block.timestamp.toI64() / SECONDS_PER_DAY).toString();
+  let financialMetrics = FinancialsDailySnapshot.load(dayId);
+  const pool = getOrCreatePool(block.number, block.timestamp);
+
+  if (!financialMetrics) {
+    financialMetrics = new FinancialsDailySnapshot(dayId);
+    financialMetrics.protocol = PROTOCOL_ID;
+
+    financialMetrics.totalValueLockedUSD = pool.totalValueLockedUSD;
+    financialMetrics.dailyTotalRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeTotalRevenueUSD = pool.cumulativeTotalRevenueUSD;
+    financialMetrics.dailyProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeProtocolSideRevenueUSD =
+      pool.cumulativeProtocolSideRevenueUSD;
+    financialMetrics.dailySupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    financialMetrics.cumulativeSupplySideRevenueUSD =
+      pool.cumulativeSupplySideRevenueUSD;
+  }
+
+  // Set block number and timestamp to the latest for snapshots
+  financialMetrics.blockNumber = block.number;
+  financialMetrics.timestamp = block.timestamp;
+
+  financialMetrics.save();
+
+  return financialMetrics;
+}
+
+export function getOrCreatePoolsDailySnapshot(
+  poolAddress: Address,
```

Comment:
unused variable

Reply:
Shouldn't the linter catch this @jaimehgb

Reply:
Don't think the linter catches when it is in a function call as a parameter. Fixing this now